src/swap/types.ts

```diff
@@ -134,14 +134,16 @@ export type SwapContextType = {
   setFromToken: (t: Token) => void;
   setToAmount: (a: string) => void;
   setToToken: (t: Token) => void;
-  swapQuoteLoadingState: SwapQuoteLoadingState;
+  setSwapLoadingState: (s: SwapLoadingState) => void;
+  swapLoadingState: SwapLoadingState;
```

Comment:
wondering if these should be optional

Reply:
what's your thinking?

Reply:
hmm i'm torn because on one hand requiring it ensures loading states are handled correctly but on the other hand making it optional allows more flexibility for users if they want to exclude

Reply:
we can make optional later if folks ask for it.